tests/jobs/scala/src/main/scala/KafkaWordCount.scala

```diff
@@ -0,0 +1,108 @@
+import java.util.{Date}
+
+import com.datastax.spark.connector._
+import com.datastax.spark.connector.cql.CassandraConnector
+import org.apache.kafka.clients.consumer.ConsumerConfig
+import org.apache.kafka.common.serialization.StringDeserializer
+import org.apache.log4j.{Level}
+import org.apache.spark.internal.Logging
+import org.apache.spark.streaming.kafka010.{ConsumerStrategies, KafkaUtils, LocationStrategies}
+import org.apache.spark.streaming.{Seconds, StreamingContext}
+
+object KafkaWordCount extends Logging {
+  def main(args: Array[String]): Unit = {
+    StreamingExamples.setStreamingLogLevels(Level.WARN)
+
+    val parser = StreamingExamples.getParser(this.getClass.getSimpleName, false).parse(args, Config())
+
+    if (parser.isEmpty) {
+      logError("Bad arguments")
+      System.exit(1)
+    }
+
+    val config = parser.get
+    println(s"Using config: $config")
+
+    val spark = StreamingExamples.createSparkSession(config)
+
+    val streamingContext = new StreamingContext(spark.sparkContext, Seconds(config.batchSizeSeconds))
+
+    // Create direct kafka stream with brokers and topics
+    val topicsSet = config.topics.split(",").toSet
+
+    val kafkaParams = getKafkaProperties(config.brokers, config.groupId, config.isKerberized)
+
+    val shouldWriteToCassandra: Boolean = config.shouldWriteToCassandra
+
+    // Set up the cassandra session:
+    val keyspaceName = config.cassandraKeyspace
+    val tableName = config.cassandraTable
+    val cassandraColumns = SomeColumns("ts" as "_1", "word" as "_2", "count" as "_3")
+
+    if (shouldWriteToCassandra) {
+      CassandraConnector(spark.sparkContext.getConf).withSessionDo { session =>
+        session.execute(s"""
+          CREATE KEYSPACE IF NOT EXISTS $keyspaceName WITH REPLICATION = {
+            'class': 'SimpleStrategy',
+            'replication_factor': 3
+          }
+        """)
+        session.execute(s"""
+          CREATE TABLE IF NOT EXISTS $keyspaceName.$tableName (
+            ts timestamp,
+            word text,
+            count int,
+            PRIMARY KEY(word, ts)
+          )
+        """)
+      }
+    }
+
+    val messages = KafkaUtils.createDirectStream[String, String](
+      streamingContext,
+      LocationStrategies.PreferConsistent,
+      ConsumerStrategies.Subscribe[String, String](topicsSet, kafkaParams)
+    )
+
+    // Get the lines, split them into words, count the words and print
+    val lines = messages.map(_.value)
+    val words = lines.flatMap(_.split(" "))
+    val wordCounts = words
+      .map(x => (x, 1L))
+      .reduceByKey(_ + _)
+      .map(x => (new Date().getTime(), x._1, x._2))
```

Comment:
One question I had was whether this was evaluated only once, or every time we perform the map. Ideally we would want the batch timestamp written to Cassandra. This might be more suited to call to `rdd.saveToCassandra below`.